test/jdk/java/net/httpclient/CancelStreamedBodyTest.java

```diff
@@ -0,0 +1,432 @@
+            }
+        }
+        assert i == uris.length * 2;
+        return result;
+    }
+
+    private HttpClient makeNewClient(boolean shared) {
+        clientCount.incrementAndGet();
+        var client = HttpClient.newBuilder()
+                .proxy(HttpClient.Builder.NO_PROXY)
+                .executor(executor)
+                .sslContext(sslContext)
+                .build();
+        return shared ? client : TRACKER.track(client);
+    }
+
+    HttpClient newHttpClient(boolean share) {
+        if (!share) return makeNewClient(share);
+        HttpClient shared = sharedClient;
+        if (shared != null) return shared;
+        synchronized (this) {
+            shared = sharedClient;
+            if (shared == null) {
+                shared = sharedClient = makeNewClient(share);
+            }
+            return shared;
+        }
+    }
+
+    final static String BODY = "Some string |\n that ?\n can |\n be split ?\n several |\n ways.";
+
+
+    @Test(dataProvider = "urls")
+    public void testAsLines(String uri, boolean sameClient)
+            throws Exception {
+        checkSkip();
+        HttpClient client = null;
+        uri = uri + "/testAsLines";
+        out.printf("%n%s testAsLines(%s, %b)%n", now(), uri, sameClient);
+        for (int i=0; i< ITERATION_COUNT; i++) {
+            if (!sameClient || client == null)
+                client = newHttpClient(sameClient);
+
+            HttpRequest req = HttpRequest.newBuilder(URI.create(uri))
+                    .GET()
+                    .build();
+            Exception failed = null;
+            List<String> lines = null;
+            for (int j = 0; j < 2; j++) {
+                try (Stream<String> body = client.send(req, BodyHandlers.ofLines()).body()) {
+                    lines = body.limit(j).toList();
+                    assertEquals(lines, BODY.replaceAll("\\||\\?", "")
+                            .lines().limit(j).toList());
+                }
+                var error = TRACKER.check(500,
+                        (t) -> t.getOutstandingOperations() > 0 || t.getOutstandingSubscribers() > 0,
+                        "subscribers for testAsLines(%s)\n\t step [%s,%s]".formatted(req.uri(), i,j),
+                        false);
+                Reference.reachabilityFence(client);
+                if (error != null) throw error;
+            }
+            if (sameClient) continue;
+            client = null;
+            System.gc();
+            var error = TRACKER.check(500);
+            if (error != null) throw error;
+        }
+    }
+
+    @Test(dataProvider = "urls")
+    public void testInputStream(String uri, boolean sameClient)
+            throws Exception {
+        checkSkip();
+        HttpClient client = null;
+        uri = uri + "/testInputStream";
+        out.printf("%n%s testInputStream(%s, %b)%n", now(), uri, sameClient);
+        for (int i=0; i< ITERATION_COUNT; i++) {
```

Comment:
Should we reconsider running each test more than once? Here it looks like we are running 3 * 2 = 6 times for testing this use case.